src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -255,16 +255,19 @@ Json::Value SotaUptaneClient::AssembleManifest() {
 }
 
 bool SotaUptaneClient::hasPendingUpdates(const Json::Value &manifests) {
+  // TODO: is this still relevant? Should use new installed_versions?
+  bool has_secondary_in_progress = false;
```

Comment:
Pretty sure that your work entirely supersedes what was there before, although that depends on how we set the `pending` flag in the secondary. I'd prefer to unify this if possible and use one canonical method for the pending state, and your new method seems like the better method to me.

Reply:
Good plan. I've debated removing OPC-UA entirely, but for now just leaving comments about things presumed to be broken is a good idea.